File: app/src/test/java/crosslanguage/KotlinServiceTest.java
package crosslanguage;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import services.kotlin.KotlinServiceApp;

public class KotlinServiceTest {
    @Test
    void testKotlinService() {
        String greeting = KotlinServiceApp.INSTANCE.greet();
        assertEquals("Hello from Kotlin Service!", greeting);
    }
}

File: KotlinService/bin/test/com/service/kotlin/KotlinServiceTest.kt
package com.service.kotlin

import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.assertEquals

class KotlinServiceTest {
    @Test
    fun testKotlinService() {
        val greeting = KotlinServiceApp.greet()
        assertEquals("Hello from Kotlin Service!", greeting)
    }
}